File: src/net/t3kt/tctrl/model/params/SingleParamModel.java
package net.t3kt.tctrl.model.params;

import java.util.Optional;
import net.t3kt.tctrl.schema.params.SingleParamSchema;

public interface SingleParamModel<S extends SingleParamSchema> extends ParamModel<S> {

    Optional<VectorParamModel<?>> getParentParam();
}

File: win/TctrlCore/Model/Params/SingleParamModel.cs
using Tctrl.Core.Schema.Params;

namespace Tctrl.Core.Model.Params {
    internal class SingleParamModel<TSchema> : ParamModel<TSchema>, ISingleParamModel where TSchema : ParamSchema {
        internal SingleParamModel(TSchema schema) : base(schema) { }
        public override bool IsSingle => true;
        public IVectorParamModel ParentParam { get; internal set; }
    }
}